### PR TITLE
Add default event filter

### PR DESCRIPTION
**Pull request by 6543**
breakout from #934

when new events are added you don't have to worry that pipeline will behave different as it does now with this

### CODE REVIEW

File: pipeline/schema/schema.json [hunk 1]
```diff
@@ -270,6 +268,10 @@
         }
       }
     },
+    "event_enum": {
+      "default": ["push", "pull_request", "tag", "deployment"],
+      "enum": ["push", "pull_request", "tag", "deployment", "cron"]
```
Comment (anbraten) [inline pipeline/schema/schema.json:273]:
```suggestion
      "enum": ["push", "pull_request", "tag", "deployment"]
```

Comment (anbraten) [reply pipeline/schema/schema.json:273]: Is it really `deployment`?

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 2]
```diff
@@ -338,9 +338,17 @@ when:
 
 #### `event`
 
+Default filter steps by event types.
+
+:::info
+**default is: `push, pull_request, tag, deployment`.**
+:::
+
+existing types: `push`, `pull_request`, `tag`, `deployment`, `cron`
```
Comment (anbraten) [inline docs/docs/20-usage/20-pipeline-syntax.md]:
```suggestion
:::info
**By default steps are filtered by following event types:**

`push, pull_request, tag, deployment`.
:::

Available types: `push`, `pull_request`, `tag`, `deployment`
```

File: pipeline/schema/schema.json (hunk 1 above)
Comment (anbraten) [reply pipeline/schema/schema.json:273]: it is. just checked it

File: docs/docs/20-usage/20-pipeline-syntax.md (hunk 2 above)
Comment (anbraten) [reply docs/docs/20-usage/20-pipeline-syntax.md]: already done ❤️

File: pipeline/schema/schema.json (hunk 1 above)
Comment (6543) [reply pipeline/schema/schema.json:273]:
```
pipeline/frontend/metadata.go
16:	EventDeploy = "deployment"

woodpecker-go/woodpecker/const.go
8:	EventDeploy = "deployment"
```

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 3]
```diff
@@ -338,9 +338,17 @@ when:
 
 #### `event`
 
+:::info
+**By default steps are filtered by following event types:**
+
+`push, pull_request, tag, deployment`.
```
Comment (anbraten) [inline docs/docs/20-usage/20-pipeline-syntax.md:344]:
```suggestion
`push`, `pull_request, `tag`, `deployment`.
```

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 4]
```diff
@@ -338,9 +338,17 @@ when:
...
+:::
+
+Available types: `push`, `pull_request`, `tag`, `deployment`, `cron`
```
Comment (anbraten) [inline docs/docs/20-usage/20-pipeline-syntax.md:347]:
```suggestion
Available types: `push`, `pull_request`, `tag`, `deployment`
```

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 5]
```diff
@@ -353,16 +361,16 @@ when:
...
 ```
 
-Execute a step for all build events:
+Execute a step for all pipeline events:
 
-```diff
+```yaml
 when:
   - event: [push, pull_request, tag, deployment]
 ```
```
Comment (anbraten) [inline docs/docs/20-usage/20-pipeline-syntax.md:376]:
```suggestion
```

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 6]
```diff
@@ -353,16 +361,16 @@ when:
...
+Execute a step for multiple events:
 
-```diff
+```yaml
 when:
   - event: [push, tag, deployment]
 ```
 
-Execute a step for all build events:
+Execute a step for all pipeline events:
```
Comment (anbraten) [inline docs/docs/20-usage/20-pipeline-syntax.md:372]:
```suggestion
```

File: docs/docs/20-usage/20-pipeline-syntax.md [hunk 7]
```diff
@@ -353,26 +361,20 @@ when:
...
 ```
 
-Execute a step for all non-pull request events:
+Execute a step for multiple events:
 
-```diff
+```yaml
 when:
   - event: [push, tag, deployment]
 ```
```
Comment (anbraten) [inline docs/docs/20-usage/20-pipeline-syntax.md:370]:
```suggestion
```

**Review by anbraten — APPROVED** [review]

**Pull request merged**